org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/backend/filebased/RepositoryUtils.java

```diff
@@ -0,0 +1,201 @@
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.LinkOption;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+import org.eclipse.winery.common.RepositoryFileReference;
+import org.eclipse.winery.common.ids.GenericId;
+import org.eclipse.winery.common.ids.Namespace;
+import org.eclipse.winery.common.ids.admin.AdminId;
+import org.eclipse.winery.common.ids.definitions.DefinitionsChildId;
+import org.eclipse.winery.common.ids.elements.ToscaElementId;
+import org.eclipse.winery.repository.Constants;
+
+class RepositoryUtils {
+
+    private final MultiRepository multiRepository;
+
+    public RepositoryUtils(MultiRepository repository) {
+        this.multiRepository = repository;
+    }
+
+    protected void checkGitIgnore() throws IOException {
+        File ignore = new File(this.multiRepository.getRepositoryDep().toFile(), Constants.FILE_GIT_IGNORE);
+
+        if (!ignore.exists()) {
+            if (ignore.createNewFile()) {
+                BufferedWriter out = new BufferedWriter(new FileWriter(ignore));
+                out.write("# Will ignore any file except the repositories.json and the local contents of the workspace folder.");
+                out.newLine();
+                out.write("/**");
+                out.newLine();
+                out.write("!/" + Constants.DEFAULT_LOCAL_REPO_NAME + "/");
+                out.newLine();
+                out.write("!/" + Constants.DEFAULT_LOCAL_REPO_NAME + "/**");
+                out.close();
+            }
+            Files.setAttribute(ignore.toPath(), "dos:hidden", true, LinkOption.NOFOLLOW_LINKS);
+        }
+    }
+
+    protected boolean checkRepoDup(String url) {
+        for (FilebasedRepository frepo : this.multiRepository.getRepositoriesMap().keySet()) {
+            if ((frepo instanceof GitBasedRepository) && ((GitBasedRepository) frepo).getRepositoryUrl().equals(url)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    protected Optional<Namespace> getNamespaceById(GenericId id) {
+
+        if (id instanceof DefinitionsChildId) {
+            return Optional.of(((DefinitionsChildId) id).getNamespace());
+        }
+
+        return Optional.empty();
+    }
+
+    protected FilebasedRepository getRepositoryByNamespace(Namespace ns) {
+        List<FilebasedRepository> repositoryList = new ArrayList<>();
+
+        for (FilebasedRepository repo : this.multiRepository.getRepositoriesMap().keySet()) {
+            for (String namespace : this.multiRepository.getRepositoriesMap().get(repo)) {
+                if (namespace.equals(ns.getDecoded())) {
+                    repositoryList.add(repo);
+                }
+            }
+        }
+
+        if (!repositoryList.isEmpty() && repositoryList.size() == 1) {
```

Comment:
the `== 1` check is somewhat restrictive considering that the following iteration does not change the returned result if the `repositoryList` was not empty before...

The second boolean condition can be dropped here